wallet/types/src/wallet_tx.rs

```diff
@@ -35,6 +37,22 @@ pub enum TxState {
     /// Transaction that is not confirmed or conflicted and is not in the mempool.
     #[codec(index = 3)]
     Inactive,
+    /// Transaction that is not confirmed or conflicted and is not in the mempool and marked as
+    /// abandoned by the user
+    #[codec(index = 4)]
+    Abandoned,
+}
+
+impl Display for TxState {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TxState::Confirmed(_, _) => f.write_str("Confirmed"),
+            TxState::Conflicted(_) => f.write_str("Conflicted"),
```

Comment:
For both Confirmed and Conflicted, can't we add the information of the arms? Like can't we instead of "Confirmed", have it as "Confirmed(1,2)"? Any reason we're not doing that?

Reply:
added more information